test/groups/upgradeableProxy/adminOperationsUpgradeableRenounceableProxy.t.sol

```diff
@@ -79,24 +127,202 @@ contract adminOperationsUpgradeableRenounceableProxy is Test, GroupSetup {
         vm.prank(group);
         proxy.renounceUpgradeability();
 
         // renounced admin
-        admin = address(uint160(uint256(vm.load(address(proxy), ADMIN_SLOT))));
+        admin = _readProxyAdminSlot();
         assertEq(admin, address(0x1));
+    }
+
+    function testNonAdminRenounceUpgradeability(address nonAdmin) public {
+        vm.assume(nonAdmin != group);
+
+        vm.prank(nonAdmin);
+        // should revert as proxy fallback after checking that caller is not admin
+        // redirects call to implementation, which doesn't have related selector
+        vm.expectRevert();
+        proxy.renounceUpgradeability();
+
+        // let's upgrade to implementation with selector clashes
+        _upgradeToAndCall(mockMintPolicyWithSelectorClashes, "");
+
+        vm.prank(nonAdmin);
+        bool returnedBool = IMockMintPolicyWithSelectorClashes(address(proxy)).renounceUpgradeability();
+        // should not call proxy native renounceUpgradeability as fallback must redirect call to the implementation,
+        // however should find a selector match inside the implementation and execute the implementation logic
+        assertTrue(returnedBool);
+        // admin shouldn't be renounced
+        address admin = _readProxyAdminSlot();
+        assertEq(admin, group);
+    }
+
+    function testRenouncedAdminEqualNonAdmin() public {
+        // todo: update forge-std
+        uint256 snapshot = vm.snapshot();
+
+        // admin should experience same behaviour as non admin after renounced upgradeability
 
-        // expect revert when trying to upgrade to implementation 0xdead
+        // default implementation
         vm.startPrank(group);
+        proxy.renounceUpgradeability();
+        vm.expectRevert();
+        proxy.renounceUpgradeability();
         vm.expectRevert();
         proxy.upgradeToAndCall(address(0xdead), "");
         vm.stopPrank();
+
+        // let's revert to initial state to upgrade to implementation with selector clashes
+        vm.revertTo(snapshot);
+        _upgradeToAndCall(mockMintPolicyWithSelectorClashes, "");
+
+        // implementation with selector clashes
+        vm.startPrank(group);
+        proxy.renounceUpgradeability();
+        bool returnedBool = IMockMintPolicyWithSelectorClashes(address(proxy)).renounceUpgradeability();
+        assertTrue(returnedBool);
+        (address returnedAddress, bytes memory returnedBytes) =
+            IMockMintPolicyWithSelectorClashes(address(proxy)).upgradeToAndCall(newMintPolicy, "newMintPolicy");
+        assertEq(returnedAddress, newMintPolicy);
+        assertEq(keccak256(returnedBytes), keccak256("newMintPolicy"));
+        vm.stopPrank();
+    }
+
+    // External receive()
+
+    function testReceive(address anyAddress) public {
+        vm.deal(anyAddress, 1 ether);
+        vm.expectRevert(UpgradeableRenounceableProxy.BlockReceive.selector);
+        (bool success,) = address(proxy).call{value: 1 ether}("");
```

Comment:
shouldn't `anyAddress` be sending value 1 eth?

Reply:
good catch! missed one line!
```suggestion
        vm.prank(anyAddress);
        (bool success,) = address(proxy).call{value: 1 ether}("");
```